flit/common.py

```diff
@@ -227,17 +227,23 @@ class Metadata:
     requires_external = ()
     provides_extra = ()
 
-    metadata_version="2.1"
+    metadata_version = "2.1"
 
-    # this is part of metadata spec 2, we are using it for installation but it
-    # doesn't actually get written to the metadata file
+    # metadata spec 2
     dev_requires = ()
+    extras_require = {}
```

Comment:
Metadata 2.0 was dropped, and metadata 2.1 is much closer to the 1.x formats. This metadata class is meant to closely represent the metadata we actually write/upload, so I'd rather add the extra requirements to `requires_dist` rather than adding a dict attribute to it.

The `_prep_metadata()` function in `flit.inifile` is responsible for converting the metadata we read from the file to a dict representing the standardised Python metadata.